nexus/db-queries/src/db/pub_test_utils/mod.rs

```diff
@@ -247,6 +247,15 @@ impl TestDatabase {
         }
     }
 
+    pub async fn extra_datastore(&self, log: &Logger) -> Arc<DataStore> {
```

Comment:
Why do we need this, as opposed to cloning the return value of `datastore()`?

Reply:
The datastore (really, the underlying pool) has in-memory state associated with being quiesced.  In our test, we want independent instances of this so that they can be quiesced independently.

Reply:
Comment added in e8eb204f96309d21e84e43feee2b09ad635d7ddd.